core/dbt/compilation.py

```diff
@@ -320,6 +320,10 @@ def _recursively_prepend_ctes(
         if model.compiled_code is None:
             raise DbtRuntimeError("Cannot inject ctes into an uncompiled node", model)
 
+        flags = get_flags()
```

Comment:
`self.config.args` is an alias to the same `flags` global object obtained via `get_flags` and is generally preferred to the global access via `get_flags` where possible. 

Similar to flags, self.config.args does not have a[ safe mechanism to access attributes that are not set in the context of the running command](https://github.com/dbt-labs/dbt-core/issues/6259). 

So: `self.config.args.suppress_ephemeral_ctes` would be unsafe in the context of a non-compiled command, but given this is part of the `Compiler` -- it feels like a relatively sane thing to do. Alternatively, we could still do `getattr(self.config.args.suppress_ephemeral_ctes, False)` as you've implemented so far.

Reply:
On second thought - `Compiler` is used in tasks that inherit from the compile task - not just the `compile` command. Which you've probably already realized :) 

Let's go with `getattr(self.config.args.suppress_ephemeral_ctes, False)` and add a comment with the tech_debt issue (#6259) for future reference/clean-up.